Sources/TokamakCore/Fiber/Fiber.swift

```diff
@@ -253,46 +239,172 @@ public extension FiberReconciler {
 
       let environment = parent?.outputs.environment ?? .init(.init())
       state = bindProperties(to: &view, typeInfo, environment.environment)
-      self.view = view
+      content = content(for: view)
       outputs = V._makeView(.init(
         content: view,
         environment: environment
       ))
 
-      visitView = { [weak self] in
-        guard let self = self else { return }
-        // swiftlint:disable:next force_cast
-        $0.visit(self.view as! V)
-      }
-
       if Renderer.isPrimitive(view) {
-        return .init(from: view, shouldLayout: reconciler?.renderer.shouldLayout ?? false)
+        return .init(from: view, useDynamicLayout: reconciler?.renderer.useDynamicLayout ?? false)
       } else {
         return nil
       }
     }
 
-    public var debugDescription: String {
-      if let text = view as? Text {
-        return "Text(\"\(text.storage.rawText)\")"
+    init<A: App>(
+      _ app: inout A,
+      rootElement: Renderer.ElementType,
+      rootEnvironment: EnvironmentValues,
+      reconciler: FiberReconciler<Renderer>
+    ) {
+      self.reconciler = reconciler
+      child = nil
+      sibling = nil
+      // `App`s are always the root, so they can have no parent.
+      parent = nil
+      elementParent = nil
+      element = rootElement
+      typeInfo = TokamakCore.typeInfo(of: A.self)
+
+      state = bindProperties(to: &app, typeInfo, rootEnvironment)
+      outputs = .init(
+        inputs: .init(content: app, environment: .init(rootEnvironment)),
+        layoutComputer: RootLayoutComputer.init
+      )
+
+      content = content(for: app)
+
+      let alternateApp = app
+      createAndBindAlternate = {
+        // Create the alternate lazily
+        let alternate = Fiber(
+          bound: alternateApp,
+          alternate: self,
+          outputs: self.outputs,
+          typeInfo: self.typeInfo,
+          element: self.element,
+          reconciler: reconciler
+        )
+        self.alternate = alternate
```

Comment:
doesn't this capture `self` strongly and create a reference cycle?

Reply:
I don't believe it does... `alternate` is a `weak` property, and its binding to a fresh `Fiber` instance with no strong reference to itself, only the parent (or the left sibling) of the alternate keeps a strong reference to it. Unless I'm overlooking something.

Reply:
Oh, unless you meant the closure itself, not the `alternate` property. I think you're right, I just added `[weak self]` to it.